pysdsl/types/intvector.hpp

```diff
@@ -19,35 +19,32 @@ namespace py = pybind11;
 
 
 template <class T,
-          unsigned int width = static_cast<unsigned int>(T::fixed_int_width)>
+          unsigned int width = static_cast<unsigned int>(T::fixed_int_width),
+          std::enable_if_t<!(width & (width - 1)) && (width & (128u - 8u))>* dummy = nullptr>
 inline auto add_int_init(py::module& m, const char* name)
 {
-    if (width == 8 || width == 16 || width == 32 || width == 64)
-    {
-        return py::class_<T>(m, name, py::buffer_protocol())
-            .def_buffer([] (T& self) {
-                char sym;
-                if (width == 8) {
-                    sym = 'B'; }
-                else if (width == 16) {
-                    sym = 'H'; }
-                else if (width == 32) {
-                    sym = 'I'; }
-                else if (width == 64) {
-                    sym = 'Q'; }
-
-                return py::buffer_info(
-                    reinterpret_cast<void*>(self.data()),
-                    width / 8,
-                    std::string(1, sym),
-                    1,
-                    { detail::size(self) },
-                    { width / 8 }
-                ); });
-    }
-    return py::class_<T>(m, name);
+    static const char syms[] = {'B', 'H', 'I', 'Q'};
+
+    return py::class_<T>(m, name, py::buffer_protocol())
+        .def_buffer([syms] (T& self) {
+            return py::buffer_info(
+                reinterpret_cast<void*>(self.data()),
+                width / 8,
+                std::string(1, syms[(unsigned int)(width / 8 - 1)]),
+                1,
+                { detail::size(self) },
+                { width / 8 }
+            ); });
 }
 
+template <class T,
+          unsigned int width = static_cast<unsigned int>(T::fixed_int_width),
+          std::enable_if_t<(width & (width - 1)) || !(width & (128u - 8u))>* dummy = nullptr>
```

Comment:
same thing with dummy

Reply:
And also, please add a comment about this `width & (width - 1)) || !(width & (128u - 8u)` stuff